File: lint-arwaky/archive/agent/git_diff_manager.py
"""Git diff result implementation (Agent Layer)."""

from __future__ import annotations
from dataclasses import dataclass
from ..contract import GitDiffResultAggregate
from ..taxonomy import Count, FilePathList, RenamedFileList


@dataclass
class GitDiffResult(GitDiffResultAggregate):
    """Concrete implementation of GitDiffResultAggregate."""

    added: FilePathList
    modified: FilePathList
    deleted: FilePathList
    renamed: RenamedFileList
    lintable_files: FilePathList
    all_files: FilePathList
    total_changed: Count

File: lint-arwaky/src-rust/agent/git_diff_manager.rs
// git_diff_manager — Git diff result implementation (Agent Layer).
use crate::contract::DiffResultAggregate;
use crate::taxonomy::{Count, FilePathList, RenamedFileList};

#[derive(Debug, Clone)]
pub struct GitDiffResult {
    pub added: FilePathList,
    pub modified: FilePathList,
    pub deleted: FilePathList,
    pub renamed: RenamedFileList,
    pub lintable_files: FilePathList,
    pub all_files: FilePathList,
    pub total_changed: Count,
}

impl DiffResultAggregate for GitDiffResult {}
